### PR TITLE
feat: 0 violation TTL for Infra Conditions returns warning

### PR DESCRIPTION
**Pull request by nathanweddle**
1. Going forward, a violation close timer (TTL) must be set to a value > 0. The default will be 24 hours. 
2. Updated the documentation as well

### CODE REVIEW

File: newrelic/resource_newrelic_infra_alert_condition.go [hunk 1]
```diff
@@ -166,8 +166,8 @@ func resourceNewRelicInfraAlertCondition() *schema.Resource {
 				Type:         schema.TypeInt,
 				Optional:     true,
 				Default:      24,
-				ValidateFunc: intInSlice([]int{0, 1, 2, 4, 8, 12, 24, 48, 72}),
-				Description:  "Determines how much time, in minutes, will pass before a violation is automatically closed. Setting the time limit to 0 prevents a violation from being force-closed. Valid values are 0, 1, 2, 4, 8, 12, 24, 48, or 72",
+				ValidateFunc: validateViolationCloseTimer(),
+				Description:  "Determines how much time, in hours, will pass before a violation is automatically closed. Valid values are 1, 2, 4, 8, 12, 24, 48, or 72",
```
Comment (kidk) [inline newrelic/resource_newrelic_infra_alert_condition.go:170]: Did the API change from minutes to hours or was the description incorrect?

File: newrelic/validation.go [hunk 2]
```diff
@@ -6,6 +6,24 @@ import (
...
+	return func(i interface{}, k string) (warnings []string, errors []error) {
+		val, ok := i.(int)
+		if !ok {
+			errors = append(errors, fmt.Errorf("expected type of %s to be int", k))
+			return
+		}
+		switch val {
+		case 1, 2, 4, 8, 12, 24, 48, 72:
+		case 0:
+			warnings = append(warnings, "0 is no longer a valid value. Using the default value of 24")
```
Comment (kidk) [inline newrelic/validation.go:19]: Where do we set the default 24 value? I'm not seeing it in the changes.

**Review by kidk — COMMENTED** [review]
Couple of questions added. Could you also remove the "terraform-provider-newrelic" binary file?

File: newrelic/resource_newrelic_infra_alert_condition.go (hunk 1 above)
Comment (nathanweddle) [reply newrelic/resource_newrelic_infra_alert_condition.go:170]: Yeah, this description was incorrect. The API has always been in hours

File: newrelic/validation.go (hunk 2 above)
Comment (nathanweddle) [reply newrelic/validation.go:19]: The default value is enforced in the API itself. We just want to give a warning here so we don't break a customer's Terraform project.

**Review by kidk — APPROVED** [review]

**Comment by mbazhlekova** [comment]
Thanks, @nathanweddle! The secrets are only in the upstream repo for the integration tests so we moved your branch there. We're continuing work here #1396. Closing this PR.

**Pull request closed without merging**